open-metadata-implementation/access-services/data-platform/data-platform-server/src/main/java/org/odpi/openmetadata/accessservices/dataplatform/handlers/RegistrationHandler.java

```diff
@@ -0,0 +1,105 @@
+
+        final String methodName = "createSoftwareServerCapability";
+
+        String qualifiedNameForSoftwareServer = softwareServerCapability.getQualifiedName();
+
+        invalidParameterHandler.validateUserId(Constants.DATA_PLATFORM_USER_ID, methodName);
+        invalidParameterHandler.validateName(qualifiedNameForSoftwareServer, QUALIFIED_NAME_PROPERTY_NAME, methodName);
+
+        InstanceProperties softwareServerProperties = new EntityPropertiesBuilder()
+                .withStringProperty(QUALIFIED_NAME_PROPERTY_NAME, qualifiedNameForSoftwareServer)
+                .withStringProperty(PATCH_LEVEL__PROPERTY_NAME, softwareServerCapability.getPatchLevel())
+                .withStringProperty(TYPE_PROPERTY_NAME, softwareServerCapability.getDataPlatformType())
+                .withStringProperty(VERSION__PROPERTY_NAME, softwareServerCapability.getDataPlatformVersion())
+                .withStringProperty(SOURCE_PROPERTY_NAME, softwareServerCapability.getSource())
+                .withStringProperty(NAME_PROPERTY_NAME, softwareServerCapability.getDisplayName())
+                .withStringProperty(Constants.DESCRIPTION, softwareServerCapability.getDescription())
+                .build();
+
+        return repositoryHandler.createEntity(
+                Constants.DATA_PLATFORM_USER_ID,
+                SOFTWARE_SERVER_CAPABILITY_TYPE_GUID,
+                SOFTWARE_SERVER_CAPABILITY_TYPE_NAME,
+                softwareServerProperties,
+                methodName);
+    }
+
+
+    /**
+     * Gets software server capability guid by qualified name .
+     *
+     * @param userId        the user id
+     * @param qualifiedName the qualified name
+     * @return the software server capability guid
+     * @throws UserNotAuthorizedException the user not authorized exception
+     * @throws PropertyServerException    the property server exception
+     */
+    public String getSoftwareServerCapabilityGuidByQualifiedName(String userId, String qualifiedName) throws InvalidParameterException,UserNotAuthorizedException, PropertyServerException {
+
+        final String methodName = "getSoftwareServerCapabilityByQualifiedName";
+
+        invalidParameterHandler.validateUserId(userId, methodName);
+        invalidParameterHandler.validateName(qualifiedName, QUALIFIED_NAME_PROPERTY_NAME, methodName);
+
+        InstanceProperties properties = repositoryHelper.addStringPropertyToInstance(
+                serviceName,
+                null,
+                QUALIFIED_NAME_PROPERTY_NAME,
+                qualifiedName,
+                methodName);
+
+        EntityDetail retrievedEntity = repositoryHandler.getUniqueEntityByName(
+                userId,
+                qualifiedName,
+                QUALIFIED_NAME_PROPERTY_NAME,
+                properties,
+                SOFTWARE_SERVER_CAPABILITY_TYPE_GUID,
```

Comment:
you could use repositoryHelper.getTypeByName() to load the typeDef for SoftwareServerCapability. In this way you don't need to duplicate the guid in your code.

Reply:
Ah great I will change it. Thx!